common.go

```diff
@@ -0,0 +1,37 @@
+package set
+
+// CommonSet is the interface that all sets implement
+type CommonSet[T any] interface {
+	Slice() []T
+	Insert(T) bool
+	InsertSlice([]T) bool
+	Size() int
```

Comment:
Let's add doc strings to each of these, since they will show up prominently in GoDocs